messaging/src/main/java/org/axonframework/messaging/responsetypes/MultipleInstancesResponseType.java

```diff
@@ -44,42 +44,88 @@
      * account. For example a {@code List<ExpectedType>} or {@code <E extends ExpectedType> List<E>}.</li>
-     * <li>If the response type is a {@link java.lang.reflect.ParameterizedType} containing a single
+     * <li>true: If the response type is a {@link java.lang.reflect.ParameterizedType} containing a single
      * {@link java.lang.reflect.WildcardType} which is assignable to the response type, taking generic types into
      * account. For example a {@code <E extends ExpectedType> List<? extends E>}.</li>
+     * <li>true: If the responseType is a single instance type matching the given {@link Type}.</li>
      * </ul>
+     * <p>
+     * If there is no match at all, it will return false to indicate a non-match.
      *
      * @param responseType the response {@link java.lang.reflect.Type} of the query handler which is matched against
-     * @return true for arrays, generic arrays and {@link java.lang.reflect.ParameterizedType}s (like a
-     * {@link java.lang.Iterable}) for which the contained type is assignable to the expected type
+     * @return true for arrays, generic arrays and {@link
+     * java.lang.reflect.ParameterizedType}s (like a {@link java.lang.Iterable}) for which the contained type is
+     * assignable to the expected type, {@link ResponseType#MATCHES_SINGLE} for matching single instances and {@link
+     * ResponseType#NO_MATCH} for non-matches
      */
     @Override
     public boolean matches(Type responseType) {
+        if (isMatchingIterable(responseType)) {
+            return true;
+        }
+        return instanceResponseType.matches(responseType);
+    }
+
+    /**
+     * Match the query handler its response {@link Type} with this implementation its responseType {@code R}. Will
+     * return a value greater than 0 in the following scenarios:
+     * <ul>
+     * <li>{@link ResponseType#MATCHES_LIST}: If the response type is an array of the expected type. For example a {@code ExpectedType[]}</li>
+     * <li>{@link ResponseType#MATCHES_LIST}: If the response type is a {@link java.lang.reflect.GenericArrayType} of the expected type.
+     * For example a {@code <E extends ExpectedType> E[]}</li>
+     * <li>{@link ResponseType#MATCHES_LIST}: If the response type is a {@link java.lang.reflect.ParameterizedType} containing a single
+     * {@link java.lang.reflect.TypeVariable} which is assignable to the response type, taking generic types into
+     * account. For example a {@code List<ExpectedType>} or {@code <E extends ExpectedType> List<E>}.</li>
+     * <li>{@link ResponseType#MATCHES_LIST}: If the response type is a {@link java.lang.reflect.ParameterizedType} containing a single
+     * {@link java.lang.reflect.WildcardType} which is assignable to the response type, taking generic types into
+     * account. For example a {@code <E extends ExpectedType> List<? extends E>}.</li>
+     * <li>1: If the responseType is a single instance type matching the given {@link Type}.</li>
```

Comment:
Why not link to the `ResponseType#MATCH_SINGLE` constant here?

Reply:
Done